src/main/java/teammates/ui/controller/InstructorCommentsPageData.java

```diff
@@ -21,35 +28,98 @@
-    public String nextPageLink;
-    public int numberOfPendingComments = 0;
+    private String instructorEmail;
+    private InstructorAttributes currentInstructor;
+    private CourseRoster roster;
+    private List<FeedbackSessionAttributes> feedbackSessions;
+    private String previousPageLink;
+    private String nextPageLink;
+    private int numberOfPendingComments = 0;
+    
+    private List<CommentsForStudentsTable> commentsForStudentsTables;
     
     public InstructorCommentsPageData(AccountAttributes account) {
         super(account);
     }
     
-    public String getGiverName(String giverEmail) {
-        InstructorAttributes instructor = roster.getInstructorForEmail(giverEmail);
-        String giverDisplay = giverEmail;
-        if (giverEmail.equals(COMMENT_GIVER_NAME_THAT_COMES_FIRST)) {
-            giverDisplay = "You";
-        } else if (instructor != null) {
-            String title = instructor.displayedName;
-            giverDisplay = title + " " + instructor.name;
+    public void init(boolean isViewingDraft, boolean isDisplayArchive, String courseId, String courseName,
+                     List<String> coursePaginationList, Map<String, List<CommentAttributes>> comments,
+                     InstructorAttributes instructor, CourseRoster roster,
+                     List<FeedbackSessionAttributes> feedbackSessions, int numberOfPendingComments) {
+        this.isViewingDraft = isViewingDraft;
+        this.isDisplayArchive = isDisplayArchive;
+        this.courseId = courseId;
+        this.courseName = courseName;
+        this.coursePaginationList = coursePaginationList;
+        this.comments = comments;
+        this.instructorEmail = instructor != null ? instructor.email : "no-email";
+        this.currentInstructor = instructor;
+        this.roster = roster;
+        this.feedbackSessions = feedbackSessions;
+        this.previousPageLink = retrievePreviousPageLink();
+        this.nextPageLink = retrieveNextPageLink();
+        this.numberOfPendingComments = numberOfPendingComments;
+    
+        setCommentsForStudentsTables();
+                                        
+    }
+
+    public Map<String, List<CommentAttributes>> getComments() {
+        return comments;
+    }
+    
+    public String getCourseId() {
+        return courseId;
+    }
+    
+    public String getCourseName() {
+        return courseName;
+    }
+    
+    public List<String> getCoursePaginationList() {
+        return coursePaginationList;
+    }
+    
+    public List<FeedbackSessionAttributes> getFeedbackSessions() {
+        return feedbackSessions;
+    }
+    
+    public Map<String, String> getGiverEmailToGiverNameMap() {
+        
+        Map<String, String> giverEmailToGiverNameMap = new HashMap<String, String>();
+        for (String giverEmail : comments.keySet()) {
+            
+            InstructorAttributes instructor = roster.getInstructorForEmail(giverEmail);
+            String giverDisplay = giverEmail;
```

Comment:
Hmm? I think I removed it from the constructor already, its just going to be used as a private method, I will change it to private